atuin-client/src/history.rs

```diff
@@ -62,10 +98,241 @@ impl History {
+    ///     .exit(0)
+    ///     .duration(100)
+    ///     .session("somesession".to_string())
+    ///     .hostname("localhost".to_string())
+    ///     .deleted_at(None)
+    ///     .env_vars(None)
+    ///     .build()
+    ///     .into();
+    /// ```
+    ///
+    /// All fields are present:
+    /// ```
+    /// use atuin_client::history::History;
+    ///
+    /// let history: History = History::from_db()
+    ///     .id("someid".to_string())
+    ///     .timestamp(chrono::Utc::now())
+    ///     .command("ls -la".to_string())
+    ///     .cwd("/home/user".to_string())
+    ///     .exit(0)
+    ///     .duration(100)
+    ///     .session("somesession".to_string())
+    ///     .hostname("localhost".to_string())
+    ///     .deleted_at(None)
+    ///     .interpreter(None)
+    ///     .context(None)
+    ///     .build()
+    ///     .into();
+    /// ```
+    pub fn from_db() -> builder::HistoryFromDbBuilder {
+        builder::HistoryFromDb::builder()
+    }
+
     pub fn success(&self) -> bool {
         self.exit == 0 || self.duration == -1
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    // @utter-step:
+    // left it here just to show that it
+    // deserializing HistoryOld into History is not a problem
+    #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, sqlx::FromRow)]
+    pub struct HistoryOld {
+        pub id: String,
+        pub timestamp: chrono::DateTime<Utc>,
+        pub duration: i64,
+        pub exit: i64,
+        pub command: String,
+        pub cwd: String,
+        pub session: String,
+        pub hostname: String,
+    }
+
+    #[test]
+    fn test_backwards_compatibility() {
+        // left it here just to show that it
+        // deserializing HistoryOld into History is not a problem with
+        // #[serde(default)] attribute set in History
+
+        #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+        pub struct HistoryOld {
+            id: String,
+            timestamp: chrono::DateTime<Utc>,
+            duration: i64,
+            exit: i64,
+            command: String,
+            cwd: String,
+            session: String,
+            hostname: String,
+        }
+
+        let history_without_delete = HistoryOld {
+            id: "test".to_string(),
+            timestamp: chrono::Utc::now(),
+            duration: 0,
+            exit: 0,
+            command: "test".to_string(),
+            cwd: "test".to_string(),
+            session: "test".to_string(),
+            hostname: "test".to_string(),
+        };
+
+        let serialized = rmp_serde::to_vec(&history_without_delete).expect("Failed to serialize");
+        let deserialized: History =
+            rmp_serde::from_slice(&serialized).expect("Failed to deserialize");
+
+        assert!(deserialized.deleted_at.is_none());
+    }
+
+    #[test]
+    #[should_panic = "Failed to deserialize: LengthMismatch(8)"]
+    fn test_forwards_compatibility_with_rmp() {
```

Comment:
But, forwards compatibility is broken with RMP – we can deserialize old object with a new client, but we cannot do same with new object and old client. The same problem was prevalent with old `History`+`HistoryWithoutDelete` approach, this test just highlights it.